### PR TITLE
sdl2 - make SDL2 pkg-config require x11, where appropriate

### LINKED ISSUE #12585: SDL2 pkg-config requires x11.
SDL2 built with X11 must have the X11 pkg-config as a require. This is the simplest reproduction:

test.c:

    #include <SDL_syswm.h>
    int main() {}

gcc test.c $(pkg-config --cflags --libs sdl2):
    
    In file included from test.c:1:0:
    /nix/store/jq3c1i7aqs22l8pycvbmqczv8w9vmghr-SDL2-2.0.3/include/SDL2/SDL_syswm.h:70:22: fatal error: X11/Xlib.h: No such file or directory
    #include <X11/Xlib.h>
                      ^
Versions:

nixos-version: 15.09.947.6783594 (Dingo)
reproduced against nixpkgs branch master / revision 7ccda42 via nix-shell.

Status: PR incoming

### PR DESCRIPTION
**Pull request by sheenobu**
Fixes #12585 

First nixpkg PR. Input is always welcome.

### CODE REVIEW

File: pkgs/development/libraries/SDL2/default.nix [hunk 1]
```diff
@@ -22,6 +22,7 @@ let
         ${if alsaSupport then "--with-alsa-prefix=${attrs.alsaLib}/lib" else ""}
         ${if (!x11Support) then "--without-x" else ""}
       '';
+  pkgConfigRequires = if (x11Support) then "x11" else "";
```
Comment (vcunat) [inline pkgs/development/libraries/SDL2/default.nix]: It would be more idiomatic to use ``stdenv.lib.optionalString x11Support "x11"``.

**Comment by vcunat** [comment]
Well, we can do such patching, but I believe it's the work of upstream to provide better *.pc files &ndash; have you reported there?

**Comment by sheenobu** [comment]
@vcunat My original thought is that x11.pc is a nixos only idiom since since X11 headers aren't in a standard location. Looking at debians libx11-dev and fedora libx11-devel, my assumption was wrong. I'll report upstream, thank you.

**Comment by sheenobu** [comment]
Reported to SDL2. Though I now wonder whether it is necessary since SDL itself doesn't /have/ to use X11, even on linux systems. We don't want to link to libX11 if the application is using directfb on linux. Might want this to be up to the library user as to whether to link to X11 libraries.

**Comment by vcunat** [comment]
Upstream link: https://bugzilla.libsdl.org/show_bug.cgi?id=3246. I'd maybe wait for them, as over-linking is a reasonable concern. It's possible that adding `--as-needed` would "do the right thing".

**Pull request closed without merging**